### ISSUE TITLE
About the dataset in train.py

### ISSUE THREAD

**Issue by twowwj** [FreedomGu/Diffportrait360#11]
Hi Diffportrait360 team,

Thanks for your code, and this is promising work.
I have been trying to reproduce the entire pipeline and tried to run "train.sh" according to the existing information from the original paper and README.md. However, it's a bit difficult to do so.

Could you give me a clear instruction to understand your "dataset/full_head_clean.py":
For example.
1. Is this part the identity index of RenderMe360?https://github.com/FreedomGu/Diffportrait360/blob/91106e7098d8fbd84a1e397033136322a969c1eb/diffportrait360_release/code/dataset/full_head_clean.py#L20C9-L22C40
2. Is this part the identity index of PanoHead?
https://github.com/FreedomGu/Diffportrait360/blob/91106e7098d8fbd84a1e397033136322a969c1eb/diffportrait360_release/code/dataset/full_head_clean.py#L28
This part needs some guidance to follow better.
3. What does the "seed" refer to?

There are so many things in this script that confuse me.
Hope to have your instructions.

**Comment by FreedomGu** [comment]
Hi, thanks for the questions!

1/That is the index of **cameras** of RenderMe360.

2/That's part of the **camera index** of PanoHead when you sample 360 by 10 degrees.

3/Seed stands for data from SphereHead sampling(since the folder starts with 'seed').

In general, I apologize that the dataloader script might be a little bit messy since it consists of three parts of datasets during my algorithm development. I recommend constructing your own dataloader if you want to train it from scratch.(You can start with Panohead/SphereHead Sampling) And also, please don't hesitate to ask if you have any questions! I am happy to answer any questions you have.

**Comment by twowwj** [comment]
Hi, super thanks for the quick reply! Really appreciate it!
Thanks for your useful suggestions. I am also planning to implement my own dataloader as well.
But I still need your further instructions.

For now, I have obtained the RenderMe360 and SphereHead datasets, both of which have been authorized by the original authors. 
However, I am now confused about how to proceed with them.

1. For **RenderMe360**, this dataset is very large, containing 500id raw data and 20id whole data. The organizational structure of each ID is as follows:
"
Data structure:
    0026 【ID】
	├─ 0026_e0.smc 【smc file of expression 0】
	├─ ...
	├─ 0026_e11.smc 【smc file of expression 11】
	├─ 0026_h0.smc 【smc file of original hairstyle】
	├─ 0026_h1_[wig index + wig color].smc 【smc file of wigs, optional】
	├─ 0026_h2.smc 【smc file of headgear, optional】
	├─ 0026_s1.smc 【smc file of speech 1】
	├─ ...
	├─ 0026_s6.smc 【smc file of speech 6】
    0041 【ID】
	├─ ...
    ... 
"
My question is, which part should I extract for training? 

2. For **SphereHead**, the original folder name is "WildHead_v1.0", which contains "align_images" (containing 00000-00061, a total of 62 image folders, each folder contains 1k images) and "dataset.json" (aligned with your *.json file).
Do you rename them to "Seed*" or what kind of operations should I do?

3. For **PanoHead**, this is the instruction link from the official [https://github.com/SizheAn/PanoHead?tab=readme-ov-file#datasets](url). In the guidance part, I have no idea how to build a relationship with the descriptions in the experimental part in diffportrait360, as the figure shows.
<img width="520" height="556" alt="Image" src="https://github.com/user-attachments/assets/1faef3dc-6368-4d17-861c-478bbab4ee6c" />

4. What's the difference between "train.sh" and "train_multiview.sh"? I noticed their dataset class are different (full_head_clean_real_data_temporal and full_head_clean_real_data_single_frame).

Sorry for too many questions, I really want to follow this work.

best

**Comment by FreedomGu** [comment]
Hello, thanks for the questions, those are really good questions. As I am not able to access MBZUAI's server to get the original code scripts for the data processing, here I will make it clearer as best I can:

1/ For **RenderMe360**, I sampled the identities randomly from the smc files with randomly selected frames. ( I lost the original format, you can try to divide by yourself, but please be aware to remove the identities with so many weird accessories. I recommend using RenderMe360 only for reference module training since currently the community doesn't have a good head camera estimation model for intrinsic and extrinsic.)

2/ For **SphereHead**, the answer is yes, I renamed it with 'seed' as the author gave me the original format that starts with 'seed'. 

3/ For both **PanoHead**,  I set a sampler to sample pitch [0,360,10], yaw [0], roll[0]( that's flat rotation with only one free angle as shown in demo) as an initial training with 1000 identities in different seeds. I really recommend you start with **Paonhead** as it's the easiest way to construct your training dataset.

4/ train.sh is for back head generation, while train_multiview.sh and train_multiview_consistency.sh is for stage1,2 stage 3 training.

**Comment by twowwj** [comment]
Hi,

First, thanks for your time, and sorry for the late reply due to some personal issues that needed to be resolved.

I spent some time collecting datasets (especially Renderme360) and made some progress.

---

### 1) Renderme360
- I have downloaded all **500 IDs** of Renderme360 and preprocessed them using the official preprocess scripts (for the **expression part**, not for **head** and **speak** parts).  
- The format of the preprocessed data is:  

<code>
0295/  
├── 0295_data.json  [contains basic information of this ID, including camera calibration parameters (intrinsic & extrinsic)]
├── images/  
│   ├── e0_cam_00_frame_000.jpg  
│   ├── e0_cam_01_frame_000.jpg  
│   ├── e0_cam_02_frame_000.jpg  
│   ├── ...  
│   ├── e0_cam_22_frame_000.jpg  
│   ├── e1_cam_00_frame_001.jpg  
│   ├── e1_cam_01_frame_001.jpg
</code>

- Question:  
  What kind of identity is considered **weird** in your evaluation?  
  - Is there any standard for filtering them out, or is it purely based on visual inspection?  
  - If possible, could you provide an example (just the index of the Renderme360 identity would be enough)?

---

### 2) SphereHead
- As mentioned in my previous reply, I have obtained the **original dataset**.

---

### 3) Panohead
- I have successfully reproduced this repo.  
- I have obtained:  
  - **FFHQ (7K)**  
  - **K-hairstyle-hqset (4K, 1024×1024)** [Link](https://psh01087.github.io/K-Hairstyle/)  
- However, I could not find the last dataset (**15K in-house large-pose images with diverse styles**).  
- If this last dataset is not necessary for your current work, please ignore this part.

### 4） Training set construction
- Because PanoHead mainly relies on FFHQ, my understanding is that I need to randomly sample 1,000 identities (front-view images) from FFHQ (As you did before), use PanoHead (with different 1000 seeds, resulting in 1000 random ids)to generate multi-view data, and then combine this with RenderMe360 for training.

---

Now I am preparing to create my **own dataloader** as you suggested.  
Do you have any **specific recommendations or best practices** regarding this process?  
Any guidance would be greatly appreciated!

best

**Comment by FreedomGu** [comment]
for renderme360 my suggestions is: remove the identity with complex accessories, i do have a list but i don't have the access to get my mbzuai machine at this moment.

for spherehead and panohead, you can directly use their checkpoint to sample identities rather than use their original dataset.

I would suggest instead of using renderme360( since we dont have a good camera esitmation model from back head at this moment), try to use the mixture dataset with panohead and spherehead first.

**Comment by twowwj** [comment]
Thanks for your reply, I will try to do it and close this issue!

**Comment by aviralchharia** [comment]
Hi @twowwj were you able to finish the dataloader code and train the Diffportrait360 model?

**Comment by FreedomGu** [comment]
https://drive.google.com/file/d/1pJ0MbSBW8Osu99vIy1xdpM1cPrpdW_se/view?usp=drive_link here is the datatraining sample feel free to try!

**Comment by aviralchharia** [comment]
Thanks @FreedomGu! Does the full dataset follow this format?
